google/cloud/BUILD.bazel

```diff
@@ -12,7 +12,17 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-package(default_visibility = ["//visibility:public"])
+# TODO(#3701) Change the default_visibility to "//visibility:private", and
+# widen visibility as needed with per-target visibility.
+package(
+    default_deprecation = """
+    These targets are not for public use and will be removed sometime around
+    July 1, 2022. Only use Bazel targets defined in the top-level //BUILD.bazel
```

Comment:
nit: 2022-07-01 is probably more readable for folks outside the US.

Reply:
Good point. I'll change to that format once we know how long to deprecate these. I'd be happy to pick 2023-01-01 for all of this to keep it simple. We could also pick a full year from now. Or something else. WDYT?